binary_search_tree/tree.py

```diff
@@ -17,41 +17,129 @@ def __init__(self):
     # Time Complexity: 
     # Space Complexity: 
     def add(self, key, value = None):
-        pass
+        if self.root == None:
+            self.root = TreeNode(key,value)
+        current = self.root
+
+        while True:
+            if self.root.key == key:
+                break
+            elif current.key > key:
+                if current.left != None:
+                    current = current.left
+                else:
+                    current.left = TreeNode(key,value)
+                    break
+            elif current.key < key:
+                if current.right != None:
+                    current = current.right
+                else:
+                    current.right = TreeNode(key,value)
+                    break
+        # return self.root
+        # if current ==
+        
 
     # Time Complexity: 
     # Space Complexity: 
     def find(self, key):
-        pass
+        current = self.root
+        while current != None:
+            if current.key == key:
+                return current.value
+            elif key < current.key:
+                current = current.left
+            else:
+                current = current.right
+        return None
 
     # Time Complexity: 
     # Space Complexity: 
     def inorder(self):
-        pass
+        def helper(node):
+            if node == None:
+                return []
+            left_part= helper(node.left) 
+            middle_part = [{"key":node.key,"value":node.value}]
+            right_part = helper(node.right)
+            result = left_part + middle_part + right_part
```

Comment:
Just a note that doing so much array concatenation is expensive rather than concatenating to the same area each time.